OCPBUGS-61465: Add a new field Mapped into openshift-test extension Image struct

dinhxuanvu opened the pull request:
The current GetMappedImageConfigs returns the map with only mapped image config. The new func GetImageConfigsWithMappedImage will return original image information along with mapped one.

#### What type of PR is this?

<!--
Add one of the following kinds:
/kind bug
/kind cleanup
/kind documentation
/kind feature

Optionally add one or more of the following kinds if applicable:
/kind api-change
/kind deprecation
/kind failing-test
/kind flake
/kind regression
-->

#### What this PR does / why we need it:

#### Which issue(s) this PR fixes:
<!--
*Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
_If PR is about `failing-tests or flakes`, please post the related issues/tests in a comment and do not use `Fixes`_*
-->
Fixes #

#### Special notes for your reviewer:

#### Does this PR introduce a user-facing change?
<!--
If no, just write "NONE" in the release-note block below.
If yes, a release note is required:
Enter your extended release note in the block below. If the PR requires additional action from users switching to the new release, include the string "action required".

For more information on release notes see: https://git.k8s.io/community/contributors/guide/release-notes.md
-->
```release-note

```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

<!--
This section can be blank if this pull request does not require a release note.

When adding links which point to resources within git repositories, like
KEPs or supporting documentation, please reference a specific commit and avoid
linking directly to the master branch. This ensures that links reference a
specific point in time, rather than a document that may change over time.

See here for guidance on getting permanent links to files: https://help.github.com/en/articles/getting-permanent-links-to-files

Please use the following format for linking documentation:
- [KEP]: <link>
- [Usage]: <link>
- [Other doc]: <link>
-->
```docs

```

dinhxuanvu commented:
@bertinatto PTAL. Let me know if this looks more like what you have in mind. Also, I assume we are gonna make changes on extension package directly to openshift-tests-extension repo instead of here?

bertinatto reviewed (commented):
This currently doesn't only set the new field, but it overrides the original struct as well.

dinhxuanvu commented:
@bertinatto @stbenjam @smg247 PTAL

bertinatto commented on openshift-hack/cmd/k8s-tests-ext/k8s-tests.go:
```diff
@@ -88,10 +88,29 @@ func main() {
+		image := convertToImage(v)
+		image.Index = int(k)
+		original.Mapped = &image
+		kubeTestsExtension.RegisterImage(original)
```
What about just iterating over the originals once, and we go, build the mapped image?

```
	originals := image.GetOriginalImageConfigs()
	mapped := image.GetMappedImageConfigs(originals, mirror)

	for k, v := range originals {
		mappedImage := convertToImage(mapped[k], k)
		image := convertToImage(v, k)
		image.Mapped = &mappedImage
		kubeTestsExtension.RegisterImage(image)
	}
```

dinhxuanvu commented:
/jira refresh

bertinatto commented on openshift-hack/cmd/k8s-tests-ext/k8s-tests.go:
```diff
@@ -88,9 +88,16 @@ func main() {
+	for k, v := range originals {
+		image := convertToImage(v, int(k))
+		mappedImage := convertToImage(mapped[k], int(k))
+		image.Mapped = &mappedImage
```
LGTM, we just need to merge the OTE changes first.

dinhxuanvu commented:
@bertinatto PTAL. Thanks.

bertinatto reviewed (commented):
Please use `<drop>` for the dependency bump and `<carry>` for the `k8s-tests.go` change. Also, the latter contains a change in `vendor/`, which I think is an oversight.

dinhxuanvu committed:
UPSTREAM: <drop>: Update openshift-tests-extension dependency

go work vendor

Signed-off-by: Vu Dinh <vudinh@outlook.com>

dinhxuanvu committed:
UPSTREAM: <carry>: Add a new field Mapped into openshift-test extension Image struct

The new field Mapped contains all the same fields of Image resource
that can be used later for openshift-tests images cmd to parse this
Mapped key. The k8s-tests-ext will construct this Mapped key from
GetMappedImageConfigs() function from upstream.

Signed-off-by: Vu Dinh <vudinh@outlook.com>
